src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -992,7 +992,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
 
         Private Function CreateBoundCatchBlockOperation(boundCatchBlock As BoundCatchBlock) As ICatchClause
             Dim handler As Lazy(Of IBlockStatement) = New Lazy(Of IBlockStatement)(Function() DirectCast(Create(boundCatchBlock.Body), IBlockStatement))
-            Dim caughtType As ITypeSymbol = Nothing ' Manual
+            Dim caughtType As ITypeSymbol = boundCatchBlock.ExceptionFilterOpt?.Type
```

Comment:
>boundCatchBlock.ExceptionFilterOpt?.Type [](start = 44, length = 40)

This doesn't look right. How filter's type is related to the type of an exception being caught?

Reply:
I got it from the master which still using old implementation before we moved to new factory model.
http://source.roslyn.io/#Microsoft.CodeAnalysis.VisualBasic/BoundTree/Statement.vb,906

Reply:
oops. you are right. used wrong one.